### PR TITLE
Add scalastyleEnabled flag to enable/disable scalastyle linting

### PR DESCRIPTION
**Pull request by ericdotdata**

### CODE REVIEW

File: src/main/scala/com/dotdata/sbt/SbtConfigPlugin.scala [hunk 1]
```diff
@@ -195,10 +195,16 @@ object SbtConfigPlugin extends AutoPlugin {
           Test / console / scalacOptions --= scalacOptionsConsoleExclusions
         )
 
+      val scalastyleSettingsDef: Def.SettingsDefinition =
+        if (scalastyleEnabled) scalastyleSettings(scalastyleExcludes)
+        else Seq(Test, Compile).flatMap{ inConfig(_) {
+          scalastyle := {}
+        }} :+ (scalastyle := {})
```
Comment (patrickgalbreath) [inline src/main/scala/com/dotdata/sbt/SbtConfigPlugin.scala:202]: This is long enough / complex enough that I'd recommend using curly braces on the if-else and going multiline.  I also think avoiding the `inConfig(_)` flatMap would be easier to read:
```suggestion
        if (scalastyleEnabled) {
          scalastyleSettings(scalastyleExcludes)
        } else {
          Seq(
            Test / scalastyle := {}, 
            Compile / scalastyle := {},
            scalastyle := {}
          )
        }
```

File: src/sbt-test/sbt-config/scalastyle-disabled/src/main/scala/default/HelloWorld.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package default
+
+object HelloWorld extends App {
```
Comment (patrickgalbreath) [inline src/sbt-test/sbt-config/scalastyle-disabled/src/main/scala/default/HelloWorld.scala:3]: Is this still needed?  I don't see it referenced in other tests but I don't know if sbt-tests need to have some App to work against.

File: src/sbt-test/sbt-config/scalastyle-disabled/src/test/scala/default/ExampleSpec.scala [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package default
+
+import collection.mutable.ArrayStack
+import org.scalatest._
+import flatspec._
+import matchers._
+
+class ExampleSpec extends AnyFlatSpec with should.Matchers {
```
Comment (patrickgalbreath) [inline src/sbt-test/sbt-config/scalastyle-disabled/src/test/scala/default/ExampleSpec.scala:8]: Is this still needed?

**Comment by patrickgalbreath** [comment]
Looks good, I appreciate the addition of tests.

File: src/sbt-test/sbt-config/scalastyle-disabled/src/test/scala/default/ExampleSpec.scala (hunk 3 above)
Comment (ericdotdata) [reply src/sbt-test/sbt-config/scalastyle-disabled/src/test/scala/default/ExampleSpec.scala:8]: its only to get the compile/test:compile to work - we have a trigger to reformat code on test runs

File: src/main/scala/com/dotdata/sbt/SbtConfigPlugin.scala (hunk 1 above)
Comment (ericdotdata) [reply src/main/scala/com/dotdata/sbt/SbtConfigPlugin.scala]: 👍

**Pull request merged**